File: src/OMS.Core/Enums/OrderStatus.cs
using System.ComponentModel;

namespace OMS.Core.Enums;

/// <summary>
/// 注文ステータス
/// </summary>
public enum OrderStatus
{
    /// <summary>
    /// 新規（未約定）
    /// </summary>
    [Description("新規")]
    NEW,

    /// <summary>
    /// 部分約定
    /// </summary>
    [Description("部分約定")]
    PARTIAL,

    /// <summary>
    /// 全約定
    /// </summary>
    [Description("全約定")]
    FILLED,

    /// <summary>
    /// 取消済み
    /// </summary>
    [Description("取消済み")]
    CANCELED,

    /// <summary>
    /// 拒否
    /// </summary>
    [Description("拒否")]
    REJECTED,

    /// <summary>
    /// 期限切れ
    /// </summary>
    [Description("期限切れ")]
    EXPIRED
}

File: backend/oms-api/src/main/java/com/oms/api/model/OrderStatus.java
package com.oms.api.model;

/**
 * 注文ステータス
 */
public enum OrderStatus {
    NEW,       // 新規
    PARTIAL,   // 一部約定
    FILLED,    // 全量約定
    CANCELED,  // キャンセル
    REJECTED,  // 拒否
    EXPIRED    // 期限切れ
}
